fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/QuerySegmentAggregator.java

```diff
@@ -205,6 +212,10 @@ protected SqlQueryData buildSystemLevelQuery(String selectRoot, String subSelect
         
         for(Map.Entry<String,Integer> resourceEntry : resourceNameMap.entrySet()) {
             String resourceTypeName =  resourceEntry.getKey();
+            // Only search the required resource types if any.
+            if (this.resourceTypes != null && !this.resourceTypes.contains(resourceTypeName)) {
```

Comment:
what's the point of this code?

Reply:
if you set the emptyList then use !this.resourceTypes.isEmpty

Reply:
resourceTypes can be empty, if it's empty, then means not a resource type specific search